crates/motionfile/src/motion_interpolator.rs

```diff
@@ -224,6 +238,22 @@ impl<T: Debug + Interpolate<f32>> MotionInterpolator<T> {
             keyframe.spline.set_initial_positions(position);
         }
     }
+
+    pub fn remaining_estimated_duration(&self) -> Duration {
+        match (
+            self.current_state.current_frame_index(),
+            self.current_state.current_frame_time_since_start(),
+        ) {
+            (Some(index), Some(time)) => Duration::saturating_sub(
```

Comment:
To fix the problem, you need to keep the subtracted time constant when you are not in the interpolation state

Reply:
Done